corehq/apps/users/tests/test_username_validation.py

```diff
@@ -0,0 +1,135 @@
+from django.test import SimpleTestCase, TestCase
+
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.users.exceptions import (
+    InvalidDomainException,
+    InvalidUsernameException,
+    UsernameAlreadyExists,
+)
+from corehq.apps.users.models import CommCareUser
+from corehq.apps.users.validation import (
+    ReservedUsernameException,
+    _check_for_reserved_usernames,
+    _ensure_username_is_available,
+    _ensure_valid_username,
+    validate_mobile_username,
+)
+
+
+class TestMobileUsernameValidation(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain = 'test-domain'
+        cls.domain_obj = create_domain('test-domain')
+        cls.addClassCleanup(cls.domain_obj.delete)
+
+    def setUp(self) -> None:
+        super().setUp()
```

Comment:
nit: type hint and super call are unnecessary
```suggestion
    def setUp(self):
```

Reply:
Ended up removing the use of `setUp` entirely so no longer relevant, but will keep in mind for the future. Thanks.